modules/react/select/lib/Select.tsx

```diff
@@ -1,129 +1,220 @@
+import {useSelectInput} from './hooks/useSelectInput';
+import {CanvasSystemIcon} from '@workday/design-assets-types';
+import {space} from '../../tokens';
 
-export interface SelectProps extends Themeable, GrowthBehavior {
-  /**
-   * The SelectOption children of the Select (must be at least two).
-   */
-  children: React.ReactElement<SelectOption> | React.ReactElement<SelectOption>[];
+export interface SelectInputProps extends ExtractProps<typeof TextInput> {
   /**
-   * If true, set the Select to the disabled state.
-   * @default false
+   * The Icon to render at the start of the input. Use this prob if your options
+   * include icons that you would like to render in the input when selected.
+   * An option was must be select in order to render and icon.
    */
-  disabled?: boolean;
-  /**
-   * The type of error associated with the Select (if applicable).
-   */
-  error?: ErrorType;
-  /**
-   * The function called when the Select state changes.
-   */
-  onChange?: (e: React.ChangeEvent<HTMLSelectElement>) => void;
-  /**
-   * The value of the Select.
-   */
-  value?: string;
+  inputStartIcon?: CanvasSystemIcon;
 }
 
-const StyledSelect = styled('select')<SelectProps & StyledType>(
-  {
-    ...type.levels.subtext.large,
-    border: `1px solid ${inputColors.border}`,
-    display: 'block',
-    backgroundColor: inputColors.background,
-    borderRadius: borderRadius.m,
-    boxSizing: 'border-box',
-    height: space.xl,
-    minWidth: 280,
-    transition: '0.2s box-shadow, 0.2s border-color',
-    padding: space.xxs,
-    margin: 0, // Fix Safari
-    MozAppearance: 'none', // FF bugfix: hide arrow so it doesn't show under ours
-    WebkitAppearance: 'none',
-    '&::placeholder': {
-      color: inputColors.placeholder,
-    },
-    '&:hover': {
-      borderColor: inputColors.hoverBorder,
-    },
-    '&:focus:not([disabled])': {
-      borderColor: inputColors.focusBorder,
-      boxShadow: `inset 0 0 0 1px ${inputColors.focusBorder}`,
-      outline: 'none',
-    },
-    '&:disabled': {
-      backgroundColor: inputColors.disabled.background,
-      borderColor: inputColors.disabled.border,
-      color: inputColors.disabled.text,
-      '&::placeholder': {
-        color: inputColors.disabled.text,
-      },
-    },
-  },
-  ({error}) => ({
-    ...errorRing(error),
-  }),
-  ({grow}) =>
-    grow && {
-      width: '100%',
-    }
+/**
+ * The SelectInput renders a custom `ComboboxInput` with additional support for keyboard navigation
+ * and interaction as defined by [WAI](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/).
+ * All valid attributes are spread to the underlying `<input>` element.
+ */
+export const SelectInput = createSubcomponent(TextInput)({
+  modelHook: useSelectModel,
+  elemPropsHook: useSelectInput,
+})<SelectInputProps>(
+  ({placeholder = 'Choose an Option', width, inputStartIcon, ...props}, Element, model) => {
+    return (
+      <InputGroup width={width}>
```

Comment:
Because the parent container wraps the input, I want to be sure to pass the width to it in case they want to expand the width of it. Everything else gets spread to the lower level input element.